### PR TITLE
RDM-8217: Codebase Improvement: classes rename in Data Store

### PR DESCRIPTION
**Pull request by kiran-yenigala-hmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-8217

### Change description ###

Class Name renaming

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ccd/data/casedetails/CaseAuditEventRepository.java [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+import uk.gov.hmcts.ccd.domain.CaseDetails;
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/data/casedetails/CaseAuditEventRepository.java:5]: if we are not removing the model package, then better to leave CaseDetails in `uk.gov.hmcts.ccd.domain.model` where all the other domain classes are

File: src/main/java/uk/gov/hmcts/ccd/data/casedetails/JurisdictionMapper.java [hunk 2]
```diff
@@ -13,7 +13,7 @@ public JurisdictionDisplayProperties toResponse(Jurisdiction jurisdiction) {
         result.setId(jurisdiction.getId());
         result.setName(jurisdiction.getName());
         result.setDescription(jurisdiction.getDescription());
-        result.setCaseTypes(jurisdiction.getCaseTypes());
+        result.setCaseTypeDefinitions(jurisdiction.getCaseTypeDefinitions());
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/data/casedetails/JurisdictionMapper.java:16]: in this particular case adding the Definitions suffix will make things a bit confusing because we are creating a display object. Possible to revert this change?

File: src/main/java/uk/gov/hmcts/ccd/data/definition/UIDefinitionGateway.java [hunk 3]
```diff
@@ -2,24 +2,24 @@
...
+import uk.gov.hmcts.ccd.domain.model.definition.WorkbasketInputFieldsDefinition;
 
 public interface UIDefinitionGateway {
 
     SearchResult getWorkBasketResult(int version, String caseTypeId);
 
     SearchResult getSearchResult(int version, String caseTypeId);
 
-    SearchInputDefinition getSearchInputDefinitions(int version, String caseTypeId);
+    SearchInputFieldsDefinition getSearchInputDefinitions(int version, String caseTypeId);
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/data/definition/UIDefinitionGateway.java:18]: shall we rename the method names too?

File: src/main/java/uk/gov/hmcts/ccd/data/definition/UIDefinitionRepository.java [hunk 4]
```diff
@@ -48,20 +48,20 @@ public SearchInputDefinition getSearchInputDefinitions(final String caseTypeId)
         return cachedUiDefinitionGateway.getWizardPageCollection(version.getVersion(), caseTypeId, eventTriggerId);
     }
 
-    public WorkbasketInputDefinition getWorkbasketInputDefinitions(final String caseTypeId) {
+    public WorkbasketInputFieldsDefinition getWorkbasketInputDefinitions(final String caseTypeId) {
         final CaseTypeDefinitionVersion version = caseDefinitionRepository.getLatestVersion(caseTypeId);
         return cachedUiDefinitionGateway.getWorkbasketInputDefinitions(version.getVersion(), caseTypeId);
     }
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/data/definition/UIDefinitionRepository.java:55]: shall we rename the method names too?

File: src/main/java/uk/gov/hmcts/ccd/data/draft/DefaultDraftGateway.java [hunk 5]
```diff
@@ -19,8 +19,8 @@
 import org.springframework.web.util.UriComponentsBuilder;
 import uk.gov.hmcts.ccd.ApplicationParams;
 import uk.gov.hmcts.ccd.data.SecurityUtils;
-import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
-import uk.gov.hmcts.ccd.domain.model.definition.DraftResponseToCaseDetailsBuilder;
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/data/draft/DefaultDraftGateway.java:23]: 👍

File: src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/external/GetStartTriggerTest.java [hunk 6]
```diff
@@ -215,22 +215,22 @@ void should404WhenEventTriggerDoesNotExist() {
         }
     }
 
-    private RequestSpecification callGetStartCaseTrigger(String caseTypeId, String eventTriggerId) {
+    private RequestSpecification callGetStartCaseTrigger(String caseTypeId, String eventId) {
```
Comment (mario-paniccia) [inline src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/external/GetStartTriggerTest.java:218]: shall we rename?

File: src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/external/GetStartTriggerTest.java [hunk 7]
```diff
@@ -215,22 +215,22 @@ void should404WhenEventTriggerDoesNotExist() {
...
 
-    private RequestSpecification callGetStartEventTrigger(String caseId, String eventTriggerId) {
+    private RequestSpecification callGetStartEventTrigger(String caseId, String eventId) {
         return asAutoTestCaseworker(FALSE)
             .get()
             .given()
             .pathParam("caseId", caseId)
-            .pathParam("triggerId", eventTriggerId)
+            .pathParam("triggerId", eventId)
             .accept(V2.MediaType.START_EVENT_TRIGGER)
```
Comment (mario-paniccia) [inline src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/external/GetStartTriggerTest.java:234]: shall we rename START_EVENT_TRIGGER?

File: src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/internal/GetUIStartTriggerTest.java [hunk 8]
```diff
@@ -84,20 +84,20 @@ void should404WhenEventTriggerDoesNotExist() {
                 .statusCode(404);
         }
 
-        private RequestSpecification callGetStartTrigger(String caseTypeId, String eventTriggerId) {
+        private RequestSpecification callGetStartTrigger(String caseTypeId, String eventId) {
```
Comment (mario-paniccia) [inline src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/internal/GetUIStartTriggerTest.java:87]: shall we rename the method?

File: src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/internal/GetUIStartTriggerTest.java [hunk 9]
```diff
@@ -84,20 +84,20 @@ void should404WhenEventTriggerDoesNotExist() {
...
             return asAutoTestCaseworker(FALSE)
                 .get()
                 .given()
                 .pathParam("caseTypeId", caseTypeId)
-                .pathParam("triggerId", eventTriggerId)
+                .pathParam("triggerId", eventId)
                 .accept(V2.MediaType.UI_START_CASE_TRIGGER)
```
Comment (mario-paniccia) [inline src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/internal/GetUIStartTriggerTest.java:93]: shall we rename UI_START_CASE_TRIGGER?

File: src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/internal/GetUIStartTriggerTest.java [hunk 10]
```diff
@@ -177,12 +177,12 @@ void should404WhenEventTriggerDoesNotExist() {
                 .statusCode(404);
         }
 
-        private RequestSpecification callGetStartEventTrigger(String caseId, String eventTriggerId) {
+        private RequestSpecification callGetStartEventTrigger(String caseId, String eventId) {
```
Comment (mario-paniccia) [inline src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/internal/GetUIStartTriggerTest.java:180]: rename method?

File: src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/internal/GetUIStartTriggerTest.java [hunk 11]
```diff
@@ -177,12 +177,12 @@ void should404WhenEventTriggerDoesNotExist() {
...
             return asAutoTestCaseworker(FALSE)
                 .get()
                 .given()
                 .pathParam("caseId", caseId)
-                .pathParam("triggerId", eventTriggerId)
+                .pathParam("triggerId", eventId)
                 .accept(V2.MediaType.UI_START_EVENT_TRIGGER)
```
Comment (mario-paniccia) [inline src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/internal/GetUIStartTriggerTest.java:186]: rename constant?

File: src/main/java/uk/gov/hmcts/ccd/data/casedetails/JurisdictionMapper.java (hunk 2 above)
Comment (mario-paniccia) [reply src/main/java/uk/gov/hmcts/ccd/data/casedetails/JurisdictionMapper.java:16]: ideally we should not do this we should convert domain objects into DTO. Something to improve in the future

File: src/main/java/uk/gov/hmcts/ccd/data/casedetails/JurisdictionMapper.java [hunk 12]
```diff
@@ -13,7 +13,7 @@ public JurisdictionDisplayProperties toResponse(Jurisdiction jurisdiction) {
         result.setId(jurisdiction.getId());
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/data/casedetails/JurisdictionMapper.java:13]: shall we rename Jurisdiction to JurisdictionDefinition?

File: src/main/java/uk/gov/hmcts/ccd/data/definition/UIDefinitionRepository.java [hunk 13]
```diff
@@ -38,30 +38,30 @@ public SearchResult getSearchResult(final String caseTypeId) {
         return cachedUiDefinitionGateway.getSearchResult(version.getVersion(), caseTypeId);
     }
 
-    public SearchInputDefinition getSearchInputDefinitions(final String caseTypeId) {
+    public SearchInputFieldsDefinition getSearchInputDefinitions(final String caseTypeId) {
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/data/definition/UIDefinitionRepository.java:41]: shall we rename the method names too?

File: src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CaseView.java [hunk 14]
```diff
@@ -22,11 +22,11 @@ public void setChannels(String[] channels) {
         this.channels = channels;
     }
 
-    public CaseViewTrigger[] getTriggers() {
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CaseView.java:25]: we should rename the method names here, otherwise it's confusing

File: src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CaseViewField.java [hunk 15]
```diff
@@ -174,25 +174,25 @@ public void setMetadata(boolean metadata) {
...
+        caseViewField.setId(caseFieldDefinition.getId());
+        caseViewField.setLabel(caseFieldDefinition.getLabel());
+        caseViewField.setFieldType(caseFieldDefinition.getFieldType());
+        caseViewField.setHidden(caseFieldDefinition.getHidden());
+        caseViewField.setHintText(caseFieldDefinition.getHintText());
+        caseViewField.setSecurityLabel(caseFieldDefinition.getSecurityLabel());
+        caseViewField.setValidationExpression(caseFieldDefinition.getFieldType().getRegularExpression());
+        caseViewField.setAccessControlLists(caseFieldDefinition.getAccessControlLists());
+        caseViewField.setValue(data.get(caseFieldDefinition.getId()));
+        caseViewField.setMetadata(caseFieldDefinition.isMetadata());
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CaseViewField.java:195]: yeah!

File: src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CaseViewFieldBuilder.java [hunk 16]
```diff
@@ -23,43 +23,43 @@ public CaseViewFieldBuilder(final CompoundFieldOrderService compoundFieldOrderSe
         this.compoundFieldOrderService = compoundFieldOrderService;
     }
 
-    public CaseViewField build(CaseField caseField, CaseEventField eventField) {
+    public CaseViewField build(CaseFieldDefinition caseFieldDefinition, CaseEventFieldDefinition eventField) {
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CaseViewFieldBuilder.java:26]: we should rename the parameter too `eventField` to `eventFieldDefinition`

File: src/main/java/uk/gov/hmcts/ccd/data/casedetails/JurisdictionMapper.java (hunk 12 above)
Comment (kiran-yenigala-hmcts) [reply src/main/java/uk/gov/hmcts/ccd/data/casedetails/JurisdictionMapper.java:13]: Missed it... Now Renamed to JurisdictionDefinition

File: src/main/java/uk/gov/hmcts/ccd/data/definition/HttpUIDefinitionGateway.java [hunk 17]
```diff
@@ -77,16 +77,16 @@ public SearchResult getSearchResult(int version, String caseTypeId) {
     }
 
     @Override
-    public SearchInputDefinition getSearchInputDefinitions(int version, String caseTypeId) {
+    public SearchInputFieldsDefinition getSearchInputDefinitions(int version, String caseTypeId) {
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/data/definition/HttpUIDefinitionGateway.java:80]: method rename please

File: src/main/java/uk/gov/hmcts/ccd/data/definition/HttpUIDefinitionGateway.java [hunk 18]
```diff
@@ -101,16 +101,16 @@ public SearchInputDefinition getSearchInputDefinitions(int version, String caseT
     }
 
     @Override
-    public WorkbasketInputDefinition getWorkbasketInputDefinitions(int version, String caseTypeId) {
+    public WorkbasketInputFieldsDefinition getWorkbasketInputDefinitions(int version, String caseTypeId) {
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/data/definition/HttpUIDefinitionGateway.java:104]: method rename please

File: src/main/java/uk/gov/hmcts/ccd/data/definition/HttpUIDefinitionGateway.java [hunk 19]
```diff
@@ -125,15 +125,15 @@ public WorkbasketInputDefinition getWorkbasketInputDefinitions(int version, Stri
     }
 
     @Override
-    public CaseTabCollection getCaseTabCollection(int version, String caseTypeId) {
+    public CaseTypeTabsDefinition getCaseTabCollection(int version, String caseTypeId) {
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/data/definition/HttpUIDefinitionGateway.java:128]: method rename please

File: src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CaseView.java [hunk 20]
```diff
@@ -22,11 +22,11 @@ public void setChannels(String[] channels) {
...
+    public CaseViewActionableEvent[] getTriggers() {
         return triggers;
     }
 
-    public void setTriggers(CaseViewTrigger[] triggers) {
+    public void setTriggers(CaseViewActionableEvent[] triggers) {
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CaseView.java:29]: better rename also method and parameter

File: src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CaseViewFieldBuilder.java [hunk 21]
```diff
@@ -23,43 +23,43 @@ public CaseViewFieldBuilder(final CompoundFieldOrderService compoundFieldOrderSe
...
-        caseField.propagateACLsToNestedFields();
+        caseFieldDefinition.propagateACLsToNestedFields();
 
         return field;
     }
 
-    private void buildFieldType(final CaseField caseField, final CaseEventField eventField, final CaseViewField field) {
-        compoundFieldOrderService.sortNestedFieldsFromCaseEventComplexFields(caseField, eventField.getCaseEventFieldComplex(), ROOT);
-        field.setFieldType(caseField.getFieldType());
+    private void buildFieldType(final CaseFieldDefinition caseFieldDefinition, final CaseEventFieldDefinition eventField, final CaseViewField field) {
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CaseViewFieldBuilder.java:48]: eventField -> eventFieldDefinition
field -> caseViewField

File: src/main/java/uk/gov/hmcts/ccd/data/casedetails/CaseAuditEventRepository.java (hunk 1 above)
Comment (kiran-yenigala-hmcts) [reply src/main/java/uk/gov/hmcts/ccd/data/casedetails/CaseAuditEventRepository.java]: reverted this to model.definition after discussion

File: src/main/java/uk/gov/hmcts/ccd/data/definition/UIDefinitionGateway.java (hunk 3 above)
Comment (kiran-yenigala-hmcts) [reply src/main/java/uk/gov/hmcts/ccd/data/definition/UIDefinitionGateway.java]: Done

File: src/main/java/uk/gov/hmcts/ccd/data/definition/UIDefinitionRepository.java (hunk 4 above)
Comment (kiran-yenigala-hmcts) [reply src/main/java/uk/gov/hmcts/ccd/data/definition/UIDefinitionRepository.java:55]: done

File: src/main/java/uk/gov/hmcts/ccd/data/draft/DefaultDraftGateway.java (hunk 5 above)
Comment (kiran-yenigala-hmcts) [reply src/main/java/uk/gov/hmcts/ccd/data/draft/DefaultDraftGateway.java]: Done

File: src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/external/GetStartTriggerTest.java (hunk 6 above)
Comment (kiran-yenigala-hmcts) [reply src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/external/GetStartTriggerTest.java]: done

File: src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/internal/GetUIStartTriggerTest.java (hunk 8 above)
Comment (kiran-yenigala-hmcts) [reply src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/internal/GetUIStartTriggerTest.java]: done

File: src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/external/GetStartTriggerTest.java (hunk 7 above)
Comment (kiran-yenigala-hmcts) [reply src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/external/GetStartTriggerTest.java]: renamed to START_EVENT

File: src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/internal/GetUIStartTriggerTest.java (hunk 9 above)
Comment (kiran-yenigala-hmcts) [reply src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/internal/GetUIStartTriggerTest.java]: Renamed

File: src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/internal/GetUIStartTriggerTest.java (hunk 11 above)
Comment (kiran-yenigala-hmcts) [reply src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/internal/GetUIStartTriggerTest.java]: renamed

File: src/main/java/uk/gov/hmcts/ccd/data/definition/UIDefinitionRepository.java (hunk 13 above)
Comment (kiran-yenigala-hmcts) [reply src/main/java/uk/gov/hmcts/ccd/data/definition/UIDefinitionRepository.java]: renamed to getSearchInputFieldDefinitions

File: src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CaseViewFieldBuilder.java (hunk 16 above)
Comment (kiran-yenigala-hmcts) [reply src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CaseViewFieldBuilder.java]: renamed

File: src/main/java/uk/gov/hmcts/ccd/data/definition/HttpUIDefinitionGateway.java (hunk 17 above)
Comment (kiran-yenigala-hmcts) [reply src/main/java/uk/gov/hmcts/ccd/data/definition/HttpUIDefinitionGateway.java]: renamed to getSearchInputFieldDefinitions

File: src/main/java/uk/gov/hmcts/ccd/data/definition/HttpUIDefinitionGateway.java (hunk 19 above)
Comment (kiran-yenigala-hmcts) [reply src/main/java/uk/gov/hmcts/ccd/data/definition/HttpUIDefinitionGateway.java]: renamed to getCaseTypeTabsCollection

File: src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CaseView.java (hunk 20 above)
Comment (kiran-yenigala-hmcts) [reply src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CaseView.java]: renamed to setActionableEvents and getActionableEvents

File: src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/internal/GetUIStartTriggerTest.java (hunk 10 above)
Comment (kiran-yenigala-hmcts) [reply src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/internal/GetUIStartTriggerTest.java]: done

File: src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CaseView.java (hunk 14 above)
Comment (kiran-yenigala-hmcts) [reply src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CaseView.java:25]: done

File: src/main/java/uk/gov/hmcts/ccd/data/definition/HttpUIDefinitionGateway.java (hunk 18 above)
Comment (kiran-yenigala-hmcts) [reply src/main/java/uk/gov/hmcts/ccd/data/definition/HttpUIDefinitionGateway.java]: done

File: src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CaseViewFieldBuilder.java (hunk 21 above)
Comment (kiran-yenigala-hmcts) [reply src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CaseViewFieldBuilder.java]: done
